src/SparkTestBase.sol

```diff
@@ -40,22 +45,134 @@ interface IExecutable {
     function execute() external;
 }
 
-abstract contract SparkTestBase is Test {
-    address internal executor;
-    address internal payload;
+abstract contract SpellRunner is Test {
+    using DomainHelpers for Domain;
+    using DomainHelpers for StdChains.Chain;
+
+    mapping(ChainId chainId => IExecutor executor) internal executors;
+    mapping(ChainId chainId => address payload)    internal payloads;
+    mapping(ChainId chainId => Domain domain)      internal domains;
+
+    Bridge internal baseBridge;
```

Comment:
`mapping(domain  => mapping(domain => bridge))` felt too overengineer-ey for two bridges and three chains, but open to change

Reply:
We can assume a hub and spoke model, so you can put the bridge inside the big struct.

Reply:
implemented in a91a65b